### PR TITLE
Ping all session ids with 1 query from the external signaling

### PR DESCRIPTION
**Pull request by nickvergessen**
Signed-off-by: Joas Schilling <coding@schilljs.com>

### CODE REVIEW

File: lib/Controller/SignalingController.php [hunk 1]
```diff
@@ -473,15 +473,17 @@ private function backendPing(array $request): DataResponse {
...
 
+		$pingSessionIds = [];
 		$now = $this->timeFactory->getTime();
 		foreach ($request['entries'] as $entry) {
-			if (array_key_exists('userid', $entry)) {
-				$room->ping($entry['userid'], $entry['sessionid'], $now);
-			} else {
-				$room->ping('', $entry['sessionid'], $now);
+			if ($entry['sessionid'] !== '0') {
+				$pingSessionIds[] = $entry['sessionid'];
```
Comment (rullzer) [inline lib/Controller/SignalingController.php:480]: So now we don't ping for each user... Is that fine? Like the update if the userid exists is different...

Comment (nickvergessen) [reply lib/Controller/SignalingController.php:480]: well before we also pinged users which didnt have a session, but if you dont have a session we dont return you as acitve user anyway anymore, so this is fine.

Comment (rullzer) [reply lib/Controller/SignalingController.php:480]: :rocket:

**Comment by nickvergessen** [comment]
Of course signaling tests fail

**Comment by nickvergessen** [comment]
```
MariaDB [oc]> EXPLAIN SELECT * FROM `oc_talk_participants` WHERE `user_id` = 'joas' AND `room_id` = 953 AND `session_id` = 'cpNkNbk7y1vvXRrobVXiBamkxoACdzI7z5e5CWjUJaNRzEN5FbIyFLmOjd8ELa9GsM6oDd+ekfGyXZiWZyaYcIfdUsxRud3/u6BVXgZcsbcd3J/bQMCqk7zJXsQ0DU+GxHp8w6K/Cp/uE2GugqAhcoG/J99O/K0yMUQV45Nl8AJO8TYSZHNQ5nW6WIYFdIoxMb4I6igU52Pyan7ED3qGgLtJLTLPODw0YNiisUpHwEUZGhlXz77Bm4x/+LloXjr';
+------+-------------+----------------------+-------+---------------------------------------+----------+---------+-------------------+------+-------+
| id   | select_type | table                | type  | possible_keys                         | key      | key_len | ref               | rows | Extra |
+------+-------------+----------------------+-------+---------------------------------------+----------+---------+-------------------+------+-------+
|    1 | SIMPLE      | oc_talk_participants | const | tp_ident,tp_room,tp_uid,tp_session_id | tp_ident | 2049    | const,const,const |    1 |       |
+------+-------------+----------------------+-------+---------------------------------------+----------+---------+-------------------+------+-------+
1 row in set (0.00 sec)

MariaDB [oc]> EXPLAIN SELECT * FROM `oc_talk_participants` WHERE `room_id` = 953 AND `session_id` = 'cpNkNbk7y1vvXRrobVXiBamkxoACdzI7z5e5CWjUJaNRzEN5FbIyFLmOjd8ELa9GsM6oDd+ekfGyXZiWZyaYcIfdUsxRud3/u6BVXgZcsbcd3J/bQMCqk7zJXsQ0DU+GxHp8w6K/Cp/uE2GugqAhcoG/J99O/K0yMUQV45Nl8AJO8TYSZHNQ5nW6WIYFdIoxMb4I6igU52Pyan7ED3qGgLtJLTLPODw0YNiisUpHwEUZGhlXz77Bm4x/+LloXjr';
+------+-------------+----------------------+------+--------------------------------+---------------+---------+-------+------+------------------------------------+
| id   | select_type | table                | type | possible_keys                  | key           | key_len | ref   | rows | Extra                              |
+------+-------------+----------------------+------+--------------------------------+---------------+---------+-------+------+------------------------------------+
|    1 | SIMPLE      | oc_talk_participants | ref  | tp_ident,tp_room,tp_session_id | tp_session_id | 1022    | const |    1 | Using index condition; Using where |
+------+-------------+----------------------+------+--------------------------------+---------------+---------+-------+------+------------------------------------+
1 row in set (0.00 sec)
```

I think this might have actually made things worse?

**Comment by nickvergessen** [comment]
Removing the check for the room id should fix things again:
```
MariaDB [oc]> EXPLAIN SELECT * FROM `oc_talk_participants` WHERE `session_id` = 'cpNkNbk7y1vvXRrobVXiBamkxoACdzI7z5e5CWjUJaNRzEN5FbIyFLmOjd8ELa9GsM6oDd+ekfGyXZiWZyaYcIfdUsxRud3/u6BVXgZcsbcd3J/bQMCqk7zJXsQ0DU+GxHp8w6K/Cp/uE2GugqAhcoG/J99O/K0yMUQV45Nl8AJO8TYSZHNQ5nW6WIYFdIoxMb4I6igU52Pyan7ED3qGgLtJLTLPODw0YNiisUpHwEUZGhlXz77Bm4x/+LloXjr';
+------+-------------+----------------------+------+---------------+---------------+---------+-------+------+-----------------------+
| id   | select_type | table                | type | possible_keys | key           | key_len | ref   | rows | Extra                 |
+------+-------------+----------------------+------+---------------+---------------+---------+-------+------+-----------------------+
|    1 | SIMPLE      | oc_talk_participants | ref  | tp_session_id | tp_session_id | 1022    | const |    1 | Using index condition |
+------+-------------+----------------------+------+---------------+---------------+---------+-------+------+-----------------------+
1 row in set (0.00 sec)
```

As we check the session for uniqueness on join globally and not per room, this should be fine.

**Pull request merged**